EIPS/eip-3300.md

```diff
@@ -0,0 +1,66 @@
+---
+eip: 3300
+title: Phase out refunds
+author: William Morriss (@wjmelements)
+discussions-to: https://ethereum-magicians.org/t/eip-3300-phase-out-refunds/5434
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-02-26
+replaces: 3298
+---
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+Phases out the `SSTORE` and `SELFDESTRUCT` gas refunds.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+This EIP would define a block when the `SSTORE` and `SELFDESTRUCT` refunds would begin to diminish.
+The refund would step linearly downward, eroding the implicit value of such refunds at an accelerating pace.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Refunds increase block elasticity, so the block gas target can exceed the number established by miners by up to 2x.
+This can cause hesitancy for miners to increase the block gas target.
+
+Refunds, tokenized or not, are valuable to their holders, especially during congestion.
+If refunds must be removed, a gradual change in their value would be less-disruptive to the gas market than sudden abolition.
+Refund consumption would proceed, especially during periods of congestion, and the refunds would be cleaned up from the state.
+Refund creation, driven by demand, would naturally diminish as the efficiency of the refunds fall.
+As the refund value approaches the activation cost, the tokens will approach zero, but in periods of congestion they will be cleaned up.
+
+This change is less work for the protocol developers than compensation and cleanup, while likely still achieving cleanup.
+
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+On the block this EIP activates, and again every 100 blocks, the gas refunds for `SELFDESTRUCT` and `SSTORE` would diminish by 1, until reaching their refund activation cost.
+When reaching their refund activation cost, refunds will be removed.
+The refund activation cost of `SSTORE` is defined to be the lowest possible opcode cost of setting a nonzero value to zero.
+The refund activation cost of `SELFDETRUCT` is defined to be the lowest possible opcode cost of `SELFDESTRUCT`.
```

Comment:
It would be valuable to spec this out a little more clearly.  Defining the refund provided for each operation as a function of block number would be the best I think.  It would also be good to include details on how this interacts with the various refunds of SSTOREs based on their previous and new values and mention how this interacts with the SSTORE counters from [EIP-1283](https://eips.ethereum.org/EIPS/eip-1283).

Reply:
The lack of clarification is deliberate because the `SSTORE` pricing seems to change with every hard fork, and might be changed again in London. But I can be more absolute.

Reply:
I see why the prior specification was insufficient; metering approaches can be complex. If 2929 happens in Berlin I'll specify that behavior too, but it currently seems to be excluded from the hard fork meta.

Reply:
https://github.com/ethereum/eth1.0-specs/blob/master/network-upgrades/berlin.md
2929 is definitely on board for Berlin.

Reply:
Okay. I thought perhaps it was removed because it wasn't in EIP-2070.